### PR TITLE
feat: Allow alternative nullable representations also for protobuf_nosr

### PR DESCRIPTION
**Pull request by lucasbru**
### Description 

We previously introduced alternative nullable representations for nullable. This change adds support for these properties also for protobuf_nosr.

In the process, we also fix the handling of `unwrapPrimitives` for protobuf_nosr which was partially copied during the creation of the protobuf_nosr format but did not have any effect.

Docs already gave the impression that this is generally supported for the protobuf format, so no change here was necessary.

Related to #9687 

### Testing done 

Tested using QTT.

### Reviewer checklist
- [x] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [x] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUFs_-_PROTOBUF_NOSR/7.4.0_1668502778749/plan.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      "topicName" : "test_topic",
+      "formats" : {
+        "keyFormat" : {
+          "format" : "KAFKA",
+          "properties" : { }
+        },
+        "valueFormat" : {
+          "format" : "PROTOBUF_NOSR",
+          "properties" : {
+            "unwrapPrimitives" : "true"
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUFs_-_PROTOBUF_NOSR/7.4.0_1668502778749/plan.json:19]: The only difference between these new historic plans and the existing ones should be the addition of this `unwrapPrimitives` property for all `PROTOBUF_NOSR` usages, right? 

Can you also confirm my understanding regarding changes to `unwrapPrimitives` and `PROTOBUF_NOSR` in this PR: it looks like `PROTOBUF_NOSR` was already set up to handle the config option but `unwrapPrimitives` was never being set, which means we were always using `unwrapPrimitives=false`. Then after the changes in this PR, we'll start setting `unwrapPrimitives=true` for all new queries, but existing queries will continue to run with `unwrapPrimitives=false`, which means the PR maintains backwards compatibility. Is this true? Might be worth running a quick manual test to ensure that we're not breaking backwards compatibility here. AFAICT from the code changes we should be fine. 

Normally I'd feel fine about just the QTT coverage but something weird seems to be going on with (at least some of) the existing QTT coverage. For example, this QTT that I'm commenting on looks like it isn't even running `PROTOBUF_NOSR` (despite what the name suggests) in its current version (https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUF/7.1.0_1633126034008/plan.json), which means it's not offering any compatibility guarantees for the `PROTOBUF_NOSR` upgrade that we're interested in.

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/protobuf.json [hunk 2]
```diff
@@ -184,6 +184,105 @@
...
+            {
+              "name": "OUTPUT",
+              "keyFormat": {"format": "KAFKA"},
+              "partitions": 4,
+              "valueFormat": {"format": "PROTOBUF_NOSR", "properties": {"unwrapPrimitives": "true", "nullableRepresentation": "wrapper"}}
+            }
+          ]
+        }
+      }
+    },
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/protobuf.json:285]: Given that this PR is also adding `unwrapPrimitives` support for `PROTOBUF_NOSR`, can you also add the equivalent of this test? https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/protobuf.json#L269

File: ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java [hunk 3]
```diff
@@ -208,14 +177,26 @@ public Map<String, String> getValueFormatProperties(final String valueFormat) {
...
+    valueSchemaId.ifPresent(id ->
+        builder.put(ConnectProperties.SCHEMA_ID, String.valueOf(id)));
+
+    return builder.build();
+  }
+
+  private void handleNullableProtobufProperty(final String valueFormat,
+      final Builder<String, String> builder, final String propertyName) {
+    if (ProtobufFormat.NAME.equalsIgnoreCase(valueFormat)
+        || ProtobufNoSRFormat.NAME.equalsIgnoreCase(valueFormat)) {
```
Comment (vcrfxia) [inline ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java:193]: Add unit tests for `PROTOBUF_NOSR`, similar to the existing ones for `PROTOBUF`? Also for CreateSourceProperties below.

**Review by vcrfxia — APPROVED** [review]
Thanks @lucasbru ! LGTM with a question and some comments inline.

File: ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUFs_-_PROTOBUF_NOSR/7.4.0_1668502778749/plan.json (hunk 1 above)
Comment (lucasbru) [reply ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUFs_-_PROTOBUF_NOSR/7.4.0_1668502778749/plan.json]: Hmmm very good point. I think I realized in the end that things were fine as they were for `unwrapPrimitives` - it only has an effect if I infer an SQL schema from an existing protobuf schema. Without using a schema registry, it is not possible to define an "existing protobuf schema". So whether `unwrapPrimitives` is enabled or not does not make a difference. Sorry for mixing this into the PR. I reverted all changes that touched `unwrapPrimitives`.

File: ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java (hunk 3 above)
Comment (lucasbru) [reply ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java:201]: Done

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/protobuf.json (hunk 2 above)
Comment (lucasbru) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/protobuf.json:285]: Done

File: ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_primitives_with_null/7.4.0_1668095424617/plan.json [hunk 4]
```diff
@@ -0,0 +1,214 @@
+{
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_primitives_with_null/7.4.0_1668095424617/plan.json:1]: Why is there a new historic plan for this test? This PR does not affect the PROTOBUF format, does it? Same question for the other two new historic plans below.

File: ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_nosr_primitives_with_null/7.4.0_1668095424764/plan.json [hunk 5]
```diff
@@ -0,0 +1,214 @@
+{
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_nosr_primitives_with_null/7.4.0_1668095424764/plan.json:1]: Why are there two new historic plan versions for this test in this PR? There should only be one, right? Same question for the other two PROTOBUF_NOSR tests in this PR.

Comment (lucasbru) [reply ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_nosr_primitives_with_null/7.4.0_1668095424764/plan.json]: hmm you are right. Not sure why they were created

File: ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_primitives_with_null/7.4.0_1668095424617/plan.json (hunk 4 above)
Comment (lucasbru) [reply ksqldb-functional-tests/src/test/resources/historical_plans/protobuf_-_protobuf_primitives_with_null/7.4.0_1668095424617/plan.json]: hmm you are right. Not sure why they were created

**Review by vcrfxia — APPROVED** [review]
LGTM assuming the build is green. Thanks @lucasbru !

**Pull request merged**